### ISSUE TITLE
Should detect nested comments

### ISSUE THREAD

**Issue by petdance** [petdance/html-lint#18]
I had a big ugly bug that HTML::Lint didn't detect.  It was like this:

```
<!---
comment comment

this should be the end of the comment but isn't

<tags>
<tags>
<tags>
<!-- new comment -->
```

IE 6 and 8 barfed on this, but Firefox had no problem with it.  HTML::Lint didn't complain.  Nor did tidy.
